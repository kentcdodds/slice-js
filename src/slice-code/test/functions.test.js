import {curry} from 'lodash'
import {snapSlice} from './helpers/utils'

const functionsFilename = require.resolve('./fixtures/functions')
const snapFunctionsSlice = curry(snapSlice)(functionsFilename)

test(
  'sum',
  snapFunctionsSlice(({sum}) => sum(1, 2))
)

test(
  'subtract',
  snapFunctionsSlice(({subtract}) => subtract(1, 2))
)

test(
  'multiply',
  snapFunctionsSlice(({multiply}) => multiply(1, 2))
)

test(
  'divide',
  snapFunctionsSlice(({divide}) => divide(1, 2))
)
