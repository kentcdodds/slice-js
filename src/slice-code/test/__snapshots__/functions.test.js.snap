exports[`test divide 1`] = `
"export { sum, subtract, multiply, divide };

function divide(a, b) {
  return a / b;
}"
`;

exports[`test multiply 1`] = `
"export { sum, subtract, multiply, divide };

function sum(a, b) {
  return a + b;
}

function multiply(a, b) {
  let product, i;
  product = 0;
  for (i = 0; i < b; i++) {
    product = sum(product, a);
  }
  return product;
}"
`;

exports[`test subtract 1`] = `
"export { sum, subtract, multiply, divide };

function sum(a, b) {
  return a + b;
}

function subtract(a, b) {
  return sum(a, -b);
}"
`;

exports[`test sum 1`] = `
"export { sum, subtract, multiply, divide };

function sum(a, b) {
  return a + b;
}"
`;
